### PR TITLE
Install/uninstall brave vpn wireguard service

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/30486

Security review: https://github.com/brave/security/issues/1280

Devops issue: https://github.com/brave/devops/pull/9787

- The service installed to `%(VersionDir)s\BraveVpnWireguardService`
image

- Added runhooks step to DEPS to download `wireguard.dll` and `tunnel.dll` built bt CI jobs:
   - https://brave-build-deps-public.s3.brave.com/brave-vpn-wireguard-dlls/brave-vpn-wireguard-nt-dlls-0.10.1.zip
   - https://brave-build-deps-public.s3.brave.com/brave-vpn-wireguard-dlls/brave-vpn-wireguard-tunnel-dlls-v0.5.3.zip

- brave://credits updated with new licenses:
image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- https://docs.google.com/document/d/18HMN-G4NAxrdaihC-ybl9-blhCqT7Vca-E1TdK2zJls/edit#bookmark=id.bwjwcxkrgey7

### CODE REVIEW

**Comment by fmarier** [comment]
In order to make the licenses of `tunnel.dll` and `wireguard.dll` appear in `brave://credits`, you can create two a directory for each of these projects in `brave/components/third_party/` and then put copy a `LICENSE` and a `README.chromium` like we did here for the tor daemon: https://github.com/brave/brave-core/tree/master/components/third_party/tor

**Comment by fmarier** [comment]
Here's what I would put in the `README.chromium` files:
```
Name: WireGuardNT SDK
URL: https://download.wireguard.com/wireguard-nt/
License: LicenseRef-WireguardNTSDK-PrebuiltBinaries-License
```
and:
```
Name: WireGuard for Windows
URL: https://github.com/WireGuard/wireguard-windows
License: MIT
```

**Comment by fmarier** [comment]
Or this if we decide to build `wireguard.dll` from source instead:
```
Name: WireGuard for the NT Kernel
URL: https://git.zx2c4.com/wireguard-nt/
License: GPL-2.0
License File: /brave/common/licenses/GPL-2.0
```

**Review by iefremov — APPROVED** [review]
chromium_src/patches lgtm

File: chromium_src/chrome/installer/setup/install_worker.cc [hunk 1]
```diff
@@ -32,17 +32,18 @@
...
-                                      const base::Version& version) {
+base::FilePath GetBraveVPNHelperPath(const base::FilePath& target_path,
+                                     const base::Version& version) {
   return target_path.AppendASCII(version.GetString())
       .Append(brave_vpn::kBraveVPNHelperExecutable);
 }
 
-bool ConfigureBraveVPNServiceAutoRestart(const base::FilePath& exe_path,
-                                         const CallbackWorkItem&) {
+bool InstallBraveVPNService(const base::FilePath& exe_path,
```
Comment (simonhong) [inline chromium_src/chrome/installer/setup/install_worker.cc:45]: nit: `kBraveVpnHelperInstall` constants name could be renamed to `kBraveVpnServiceInstall`

**Review by simonhong — APPROVED** [review]
++

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.cc [hunk 2]
```diff
@@ -8,11 +8,85 @@
 #include <guiddef.h>
 
 #include "base/containers/cxx20_erase.h"
+#include "brave/components/brave_vpn/common/buildflags/buildflags.h"
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.cc:11]: yep, needed for channels definitions

[the remark this answers is not in the document]
Comment (simonhong) [reply components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.cc:11]: Ah right., I was mistaken that it came from `build/build_config.h` :)

**Comment by bsclifton** [comment]
Licensing information added in https://github.com/brave/brave-core/pull/18815

**Review by bsclifton — APPROVED** [review]
Changes look great! 😄👍

**Pull request merged**